common/lib/xmodule/xmodule/modulestore/inheritance.py

```diff
@@ -233,6 +234,33 @@ class InheritanceMixin(XBlockMixin):
         scope=Scope.settings
     )
 
+    @property
+    def close_date(self):
+        """
+        Return the date submissions should be closed from.
+        """
+        due_date = self.due
+
+        if self.graceperiod is not None and due_date:
+            return due_date + self.graceperiod
+        else:
+            return due_date
+
+    def is_past_due(self):
+        """
+        Is it now past this problem's due date, including grace period?
+        """
+        return (self.close_date is not None and
+                timezone.now() > self.close_date)
+
+    def has_deadline_passed(self):
+        """
+        Verifies if the problems can be submitted or not.
```

Comment:
`has_deadline_passed` this method has nothing to do with submission and decision on submission, in this context, this method is only calculating if the problems' submission deadline is passed or not. Submission decision is taken by some other method which this method doesn't know about.